### PR TITLE
Fixed an issue 

### PR DESCRIPTION
**Pull request by carcam**
Fixed an issue with Modals in BS 2 in which they do not show if modal has "fade" class as stated here:

https://stackoverflow.com/questions/19662896/twitter-bootstrap-modal-event-not-fired-when-modal-is-shown/26784112

The forum thread with this issue is:

https://www.jevents.net/discussions/blank-in-modal-displayed-by-select-location-button?start=0#reply-209622

Also fixed readonly fix for calendars not being correctly applied

### CODE REVIEW

File: component/install.php [hunk 1]
```diff
@@ -522,19 +522,15 @@ private function updateTables() {
...
                     $sql = "ALTER TABLE #__jevents_vevent ADD INDEX evaccess (access)";
                     $db->setQuery($sql);
                     @$db->execute();
-                    
-                    // Return to old value
-                    $db->setQuery("SET SESSION sql_mode=(".$db->quote($sql_mode).")");
-                    @$db->execute();                    
+                    $db->setQuery("SET SESSION sql_mode=(SELECT CONCAT(@@sql_mode,',NO_ZERO_DATE'))");
+                    @$db->execute();
+                    $db->setQuery("SET SESSION sql_mode=(SELECT REPLACE(@@sql_mode,'NO_ZERO_DATE',''))");
```
Comment (GeraintEdwards) [inline component/install.php]: Why are you changing this?

Comment (carcam) [reply component/install.php]: It was not my intenttion. It slipped into the PR probably from an out of date repo. Not sure why though as I updated my code before starting working on the fix...

Anyway as far as I can remember that code was changed due to some issues in the sql_mode detection and I remember I worked into that fix. Is my code removing the fix or is it that you didn't merged my fix?

Comment (carcam) [reply component/install.php]: I have removed the changes on install script. Please review this PR as it contains a couple of fixes that should be in next release.

Best!!

File: libraries/jevmodal/media/js/jevmodal.js [hunk 2]
```diff
@@ -125,7 +127,7 @@ function addJevModalHtml (id){
 			+'</div>';
 		}
 		else {
-			myModal = '<div class="modal  hide fade ' + modalsize + ' jevmodal" id="'+id+'" tabindex="-1" role="dialog" aria-labelledby="'+id+'Label" aria-hidden="true" >'
```
Comment (GeraintEdwards) [inline libraries/jevmodal/media/js/jevmodal.js:128]: @carcam We use 'fade' class in  other places event repeat save popup dialog, event edit modal etc.  Why is this situation different?

Comment (carcam) [reply libraries/jevmodal/media/js/jevmodal.js:128]: To be honest, this is a change from 2019. I was able to reproduce the issue in my development box so I see it like a generic problem. If you want I can reproduce the issue again and do some research again.

But the main problem was that JS was not triggering. I'm a bit surprised the user didn't come back with all latest releases...

**Pull request closed without merging**